src/transformers/modeling_flax_pytorch_utils.py

```diff
@@ -154,8 +174,25 @@ def convert_pytorch_state_dict_to_flax(pt_state_dict, flax_model):
                     f"{random_flax_state_dict[flax_key].shape}, but is {flax_tensor.shape}."
                 )
 
-        # also add unexpected weight so that warning is thrown
-        flax_state_dict[flax_key] = jnp.asarray(flax_tensor)
+        # add batch stats if the model contains batchnorm layers
+        if "params" in flax_model.params and "batch_stats" in flax_model.params:
```

Comment:
Again here I wonder if it would be cleaner to always set:
```python
flax_state_dict[flax_key] = jnp.asarray(flax_tensor)
```
And then check if `"batch_stats"` is in `flax_model.params`:
```python
if "batch_stats" in flax_model.params:
    if "mean" in flax_key[-1]:
        ...
```

Reply:
Completely agree, made the changes in the most recent commit.